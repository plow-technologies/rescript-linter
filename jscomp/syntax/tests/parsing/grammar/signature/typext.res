// note: we overparse Pext_rebindg
type t += Foo
type Foo.Bar.t += Foo
type t<'a, 'b> += Foo('a)
type Foo.Bar.t<'a, 'b> += Foo('a)

type t += private Foo
type t += private Foo | Bar
type t += private | Foo
type t += private | Foo | Bar

type t += Circle(string, int) | Point({x: int, y: int})
type t += | Circle(string, int) | Point({x: int, y: int})

type t += Foo = Bar
type t += Foo = Bar | Circle = Circle2
type t += Foo = Bar | Circle = Geo.Circle2
type t += private Foo = Bar
type t += private Foo = Bar | Circle = Circle2
type t += private Foo = Bar | Circle = Geo.Circle2
type t += | Foo = Bar
type t += | Foo = Bar | Circle = Circle2
type t += | Foo = Bar | Circle = Geo.Circle2
type t += private | Foo = Bar
type t += private | Foo = Bar | Circle = Circle2
type t += private | Foo = Bar | Circle = Geo.Circle2
type t += Foo = Geo.Metry.Bar
