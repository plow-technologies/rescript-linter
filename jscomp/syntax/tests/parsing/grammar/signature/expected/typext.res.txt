type t +=  
  | Foo 
type Foo.Bar.t +=  
  | Foo 
type ('a,'b) t +=  
  | Foo of 'a 
type ('a,'b) Foo.Bar.t +=  
  | Foo of 'a 
type t += private  
  | Foo 
type t += private  
  | Foo 
  | Bar 
type t += private  
  | Foo 
type t += private  
  | Foo 
  | Bar 
type t +=  
  | Circle of string * int 
  | Point of {
  x: int ;
  y: int } 
type t +=  
  | Circle of string * int 
  | Point of {
  x: int ;
  y: int } 
type t +=  
  | Foo = Bar
type t +=  
  | Foo = Bar
  | Circle = Circle2
type t +=  
  | Foo = Bar
  | Circle = Geo.Circle2
type t += private  
  | Foo = Bar
type t += private  
  | Foo = Bar
  | Circle = Circle2
type t += private  
  | Foo = Bar
  | Circle = Geo.Circle2
type t +=  
  | Foo = Bar
type t +=  
  | Foo = Bar
  | Circle = Circle2
type t +=  
  | Foo = Bar
  | Circle = Geo.Circle2
type t += private  
  | Foo = Bar
type t += private  
  | Foo = Bar
  | Circle = Circle2
type t += private  
  | Foo = Bar
  | Circle = Geo.Circle2
type t +=  
  | Foo = Geo.Metry.Bar